tests/integration/add_snippets_spec.lua

```diff
@@ -193,4 +193,73 @@ describe("add_snippets", function()
 			{2:-- INSERT --}                                      |]],
 		})
 	end)
+
+	it("add autosnippets by option", function()
+		exec_lua("ls.config.setup({ enable_autosnippets = true })")
+		exec_lua([[
+			ls.add_snippets("all", {
+				ls.snippet({trig="triA", autotriggered=true}, {ls.text_node("helloAworld")}, {})
+			}, {
+				key = "a"
+			} )
+		]])
+		exec_lua([[
+			ls.add_snippets("all", {
+				ls.snippet({trig="triB", autotriggered=false}, {ls.text_node("helloBworld")}, {})
+			}, {
+				key = "b"
+			} )
+		]])
+		exec_lua([[
+			ls.add_snippets("all", {
+				ls.snippet({trig="triC", autotriggered=nil}, {ls.text_node("helloCworld")}, {})
+			}, {
+				key = "c"
```

Comment:
Tests look good :+1:
I think it would be a bit nicer to add `type="snippets"` explicitly in this call, and one more tests with `autotriggered`/(`snippetType`)`=nil` and add-snippets-`type` `"autosnippet"` (to make sure the `nil` is correctly handled for both `type`s)

Reply:
Oh, right, and one test to ensure we keep behaviour for conflicting (snippet)types

Reply:
Ah alright :+1: